### ISSUE TITLE
eventfd通知丢失的可能

### ISSUE THREAD

**Issue by ballbillow** [chenshuo/muduo#173]
往eventfd里面多次write，不一定就会有多次的read事件触发吧。
比如主线程中有多个连接同时到达，假如这里面有3个是被分配到1号工作线程，而该工作线程并不一定能保证会触发到三次read事件，我实验了一下libevent，只会触发一次read事件。
实验代码大致如下：
	void EventCallback(evutil_socket_t fd, short event, void* args)
	{
		std::cout << "recv event: " << event << " from fd: " << fd << std::endl;
		uint64_t one;
		int ret = read(fd, &one, sizeof one);
		if (ret != sizeof one)
		{
			std::cout << "read evfd fail: " << ret << std::endl;
			return;
		}
		std::cout << "read from evfd: " << one << std::endl;
	}

	int ev_fd = eventfd(0, EFD_NONBLOCK|EFD_CLOEXEC);
	if (ev_fd > 0)
	{
		event_base* eventbase1 = event_base_new();
		event* ev_fd_event = event_new(eventbase1, ev_fd, EV_READ|EV_PERSIST, &EventCallback, NULL);
		event_add(ev_fd_event, NULL);
		auto thread1 = std::thread([&]()->void {event_base_dispatch(eventbase1);});
		thread1.detach();

		std::cout << "new eventfd: " << ev_fd << std::endl;
		for (size_t i=0; i<3; ++i)
		{
			uint64_t one = 1;
			int ret = write(ev_fd, &one, sizeof one);
			if (ret != sizeof one)
			{
				std::cout << "write evfd fail: " << ret << std::endl;
			}
		}
	}
粗略看了下muduo代码，应该也存在这个问题。

**Comment by ballbillow** [comment]
EFD_SEMAPHORE看来可以解决这个问题。

**Comment by ballbillow** [comment]
仔细看了下，muduo里面虽然存在多次write，只触发一次read事件，但是一次性都处理了。
doPendingFunctors，呵呵。
